src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/DotNetCoreProjectCompatibilityDetector.cs

```diff
@@ -0,0 +1,317 @@
+        /// of projects. This event always fires after the initial opening of a solution 
+        /// </summary>
+        public int OnAfterBackgroundSolutionLoadComplete()
+        {
+            // Schedule this to run on idle
+            ThreadHandling.Value.JoinableTaskFactory.StartOnIdle(async () =>
+            {
+                // Run on the background
+                await TaskScheduler.Default;
+
+                CompatibilityLevel finalCompatLevel = CompatibilityLevel.Supported;
+                IProjectService projectService = ProjectServiceAccessor.Value.GetProjectService();
+                IEnumerable<UnconfiguredProject> projects = projectService.LoadedUnconfiguredProjects;
+                foreach (var project in projects)
+                {
+                    // Track the most severe compatibility level
+                    CompatibilityLevel compatLevel = await GetProjectCompatibilityAsync(project).ConfigureAwait(false);
+                    if(compatLevel != CompatibilityLevel.Supported && compatLevel > finalCompatLevel)
+                    {
+                        finalCompatLevel = compatLevel;
+                    }
+                }
+            
+                if (finalCompatLevel != CompatibilityLevel.Supported)
+                {
+                    // Warn the user.
+                    await WarnUserOfiIncompatibleProjectAsync(finalCompatLevel).ConfigureAwait(false);
+               }
+
+               // Used to know when to process newly added projects
+               _solutionOpened = true;
+            });
+
+            return VSConstants.S_OK;
+        }
+
+        private async Task WarnUserOfiIncompatibleProjectAsync(CompatibilityLevel compatLevel)
+        {
+            // Warn the user.
+            await ThreadHandling.Value.SwitchToUIThread();
+
+            // Check if already warned - this could happen in the off chance two projects are added very quickly since the detection work is 
+            // scheduled on idle.
+            if(_compatibilityLevelWarnedForThisSolution < compatLevel)
+            {
+                // Only want to warn once per solution
+                _compatibilityLevelWarnedForThisSolution = compatLevel;
+                
+                IVsUIShell uiShell = _serviceProvider.GetService<IVsUIShell, SVsUIShell>();
+                uiShell.GetAppName(out string caption);
+
+                if(compatLevel == CompatibilityLevel.Partial)
+                {
+                    // Get current dontShowAgain value
+                    var settingsManager = (ISettingsManager)_serviceProvider.GetService(typeof(SVsSettingsPersistenceManager));
+                    bool suppressPrompt = false;
+                    if( settingsManager != null)
+                    {
+                        suppressPrompt = settingsManager.GetValueOrDefault(SuppressDotNewCoreWarningKey, defaultValue:false);
```

Comment:
Is there a way for the user to reset the value of this if they change their mind?

Reply:
Not easily since this goes into the settings store. They would likely need to delete the files under local appdata  or wherever VS stores the settings

Reply:
I thought it was a UX requirement to have a way to reset the state of options like this (usually from Tools\Options).